openai/openai.go

```diff
@@ -86,6 +92,38 @@ type ChatCompletionChunk struct {
 	Choices           []ChunkChoice `json:"choices"`
 }
 
+type CompletionRequest struct {
+	Model            string   `json:"model"`
+	Prompt           string   `json:"prompt"` // openai also accepts a list of strings if we want to do that
+	FrequencyPenalty *float64 `json:"frequency_penalty"`
+	MaxTokens        *int     `json:"max_tokens"`
+	PresencePenalty  *float64 `json:"presence_penalty"`
+	Seed             *int     `json:"seed"`
+	Stop             any      `json:"stop"`
```

Comment:
Instead of any here, let's do something like this:

```
type StringOrStringSlice struct {
	Values []string
}

// UnmarshalJSON implements the json.Unmarshaler interface.
func (s *StringOrStringSlice) UnmarshalJSON(data []byte) error {
	if err := json.Unmarshal(data, &single); err == nil {
		s.Values = []string{single}
		return nil
	}

	var slice []string
	if err := json.Unmarshal(data, &slice); err == nil {
		s.Values = slice
		return nil
	}

	return fmt.Errorf("invalid data format for StringOrStringSlice: %s", string(data))
}

type ExampleStruct struct {
	Field StringOrStringSlice `json:"field"`
}
```

Reply:
Actually, below is less code and more straightforward